cmd/zfs/zfs_main.c

```diff
@@ -8672,7 +8672,6 @@ main(int argc, char **argv)
 	int i = 0;
 	const char *cmdname;
 	char **newargv;
-	extern char **environ;
```

Comment:
This reminds me of aaa9a6700f98a62386fcdcb8e89cab884d616c5d. It is odd that people keep putting interesting declarations into `main()`.

That said, I would prefer it if we followed the standard's recommendation here:

> Applications should use the [getenv()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/getenv.html) function rather than accessing the environment directly via either envp or environ.

https://pubs.opengroup.org/onlinepubs/9699919799/functions/environ.html

Reply:
Interestingly, glibc puts `environ` behind `#ifdef __USE_GNU` in `unistd.h`. However, it has a `__environ` that can be used without specifying `__USE_GNU`. That is something that the Linux Standard Base appears to have invented:

https://refspecs.linuxbase.org/LSB_5.0.0/LSB-Core-generic/LSB-Core-generic/baselib---environ.html

It is portable across both glibc and musl libc, but it is not portable to non-Linux systems. Both illumos-gate and FreeBSD do not support it.

Reply:
If I understand your suggestion, that would require factoring out the `**environ` out of the `zfs_setproctitle_init()` function signature - is that correct?

Then `**environ` would have to be exposed to lib/libzutil/os/linux/zutil_setproctitle.c` through some fashion?

Any ideas/preference?

Reply:
I misremembered what `getenv()` does. On second thought, lets just go with this PR as is.